### PR TITLE
02 - Script to parser SDK python tests

### PR DESCRIPTION
**Pull request by hiltonlima**
The purpose of this PR is to resolve the display of the names of the python test steps.

Now we will leave it up to the SDK to deliver the information for each test. The backend will only pass the name of the test to the frontend.

The information is obtained by calling the get_test_info() method. Here is an example of the information for the `TC_ACE_1_2` test:
```
{
        "class_name": "TC_ACE_1_2",
        "desc": "test_TC_ACE_1_2",
        "function": "test_TC_ACE_1_2",
        "path": "sdk/TC_ACE_1_2",
        "pics": [],
        "steps": [
            {
                "description": "Run entire test",
                "expectation": "",
                "is_commissioning": false,
                "test_plan_number": 1
            }
        ]
    },
```

# Issue
https://github.com/project-chip/certification-tool/issues/484

With this change no changes to the frontend will be necessary. And we can close this issue.

# Code changes
- Added parsed file for python sdk tests
- Added script to generate to parse python files
- Added '--get_test_info' parameter into 'test_harness_client.py' @rquidute 

# Output report
```
###########################################################################
###############################   REPORT   ################################
###########################################################################
>>>>>>>> Output JSON file: /app/backend/test_collections/matter/sdk_tests/python_tests_17b1a38e909e7874593bcb87c31be03a5866f1d4.json
>>>>>>>> Total of test functions: 185
>>>>>>>> Total of test functions (doesn't start with 'test_TC_'): 12
>>>>>>>> Total of scripts with error: 1
Error 0: Failed running command: ['sdk/TC_CCTRL_2_3', 'TC_CCTRL_2_3', '--get_test_info']
###########################################################################
```

### CODE REVIEW

**Review by antonio-amjr — APPROVED** [review]

File: test_collections/matter/sdk_tests/support/python_testing/models/rpc_client/test_harness_client.py [hunk 1]
```diff
@@ -41,15 +42,31 @@ def main() -> None:
...
+                    run_test(
+                        script_path=sys.argv[1], class_name=sys.argv[2], config=config
+                    )
+
+
+def get_test_info(script_path: str, class_name: str, config: MatterTestConfig):
+    module = importlib.import_module(script_path.replace("/", "."))
+    TestClassReference = getattr(module, class_name)
+    v = get_test_info_support(TestClassReference, config)
+    return v
```
Comment (rquidute) [inline test_collections/matter/sdk_tests/support/python_testing/models/rpc_client/test_harness_client.py:69]:
```suggestion
    return get_test_info_support(TestClassReference, config)
```

File: test_collections/matter/sdk_tests/support/python_testing/list_python_tests_classes.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+            command_string,
+            prefix=CONTAINER_TH_CLIENT_EXEC,
+        )
+        if result.exit_code != 0:
+            errors_found.append(f"Failed running command: {command}")
+            continue
+
+        with open(JSON_OUTPUT_FILE_PATH, "r") as json_file:
+            json_data = json.load(json_file)
+            json_file.close()
```
Comment (rquidute) [inline test_collections/matter/sdk_tests/support/python_testing/list_python_tests_classes.py:102]:
```suggestion
            
```
I think when using `with` statement, it not necessary to close the file. Please check this information.

**Review by fabiowmm — CHANGES_REQUESTED** [review]
What's our end goal here? Let's clearly state our goals for the PRs we open. It may be obvious for the author but not so for the reviewers.

File: test_collections/matter/sdk_tests/support/python_testing/list_python_tests_classes.py (hunk 2 above)
Comment (antonio-amjr) [reply test_collections/matter/sdk_tests/support/python_testing/list_python_tests_classes.py:107]: You're correct Romulo, it's not necessary to close.
It's the purpose of the `with` statement indeed.

File: test_collections/matter/sdk_tests/support/python_testing/sdk_python_tests.py [hunk 3]
```diff
@@ -93,28 +100,26 @@ def _parse_python_script_to_test_case_declarations(
...
-            else:
-                if python_test_type == PythonTestType.COMMISSIONING:
-                    suites[SuiteType.COMMISSIONING].add_test_case(test_case)
-                elif python_test_type == PythonTestType.NO_COMMISSIONING:
-                    suites[SuiteType.NO_COMMISSIONING].add_test_case(test_case)
-                elif python_test_type != PythonTestType.MANDATORY:
-                    suites[SuiteType.LEGACY].add_test_case(test_case)
+    for test_case in test_cases:
+        python_test_type = test_case.class_ref.python_test.python_test_type
+        if mandatory:
```
Comment (ccruzagralopes) [inline test_collections/matter/sdk_tests/support/python_testing/sdk_python_tests.py:113]: This if statement seems redundant. The test case already has the information in `python_test_type`. Maybe this parameter is not needed in this method

**Review by ccruzagralopes — COMMENTED** [review]
Please remember to fix the CI issues

File: test_collections/matter/sdk_tests/support/python_testing/list_python_tests_classes.py [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+import json
+from pathlib import Path
+
+from test_collections.matter.config import matter_settings
+from test_collections.matter.sdk_tests.support.models.sdk_test_folder import (
+    SDKTestFolder,
+)
+from test_collections.matter.sdk_tests.support.sdk_container import SDKContainer
+
+# Make these constants synced with "test_harness_client.py"
```
[the remark this answers is not in the document]
Comment (hiltonlima) [reply test_collections/matter/sdk_tests/support/python_testing/list_python_tests_classes.py:28]: it's a permanent reminder :)

File: test_collections/matter/sdk_tests/support/python_testing/sdk_python_tests.py (hunk 3 above)
Comment (hiltonlima) [reply test_collections/matter/sdk_tests/support/python_testing/sdk_python_tests.py:113]: It's not... But we need to improve the process to perform only one pass over the tests and separate the collections correctly.

Currently, the tests are parsed twice to do this (mandatory collection and Python collection).

We can improve it, yes. But I think it's better to do this in another PR.

**Review by fabiowmm — APPROVED** [review]
Good job.

**Comment by hiltonlima** [comment]
LGTM

**Pull request merged**